using the full entity in the creation and migration apis

SrdjanStevanetic opened the pull request.

SrdjanStevanetic committed:
using the full entity in the creation and migration apis

SrdjanStevanetic committed:
validation of the body entity in the creation and update apis

gsergiu commented on entity-management-definitions/src/main/java/eu/europeana/entitymanagement/utils/EntityObjectFactory.java:
```diff
@@ -88,11 +88,35 @@
-      // entityClass cannot be null here as map contains all possible types
-      return entityClass.getDeclaredConstructor(entityClass).newInstance(entity);
-
+      switch (EntityTypes.valueOf(entity.getType())) {
```
I don't think that this code is required, the previous implementation worked just fine

SrdjanStevanetic replied:
It is required. In the previous case, when invoing entityClass.getDeclaredConstructor(entityClass).newInstance(entity), the constructor public Entity(Entity copy), was invoked, even though the entity objects were the derived ones like Organization, Place, etc. The argument to the constructor should be casted to Place, Organization, etc., so that the rigth constructor is invoked, like e.g. public Organization(Organization copy).

gsergiu commented on entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java:
```diff
@@ -447,13 +452,16 @@ public EMController(
     }
-    String creationRequestId = entityCreationRequest.getId();
+    
+    validateBodyEntity(europeanaProxyEntity);
```
doesn't the @Valid annotation trigger the validation automatically? The validation should not be called twice, if the validation needs to be performed through explicit call, the @Valid should be removed

SrdjanStevanetic replied:
The @Valid is removed. The validation needs to be called explicitelly.

gsergiu commented on entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java:
```diff
@@ -710,4 +718,20 @@ private String getDatabaseIdentifier(String entityId) {
   }
+  
+  private void validateBodyEntity (Entity entity) throws HttpBadRequestException {
+	Set<ConstraintViolation<Entity>> violations =
```
this method will need to be updated with the new changes from the specs

SrdjanStevanetic replied:
Done. I have wrote soveral comments in the code for the understandability purposes. Now, the validation should work fine for any nested fields and objects, without the need to validate separatelly those objects, only the right definition of the objects fields is needed.

gsergiu commented on entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EntityAdminController.java:
```diff
@@ -123,7 +123,7 @@ public EntityAdminController(
       @PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
       @PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
-      @RequestBody EntityPreview entityCreationRequest,
+      @RequestBody Entity europeanaProxyEntity,
```
this input needs to be validated as well

SrdjanStevanetic replied:
Improved.

gsergiu commented on entity-management-web/src/main/resources/build.properties:
```diff
@@ -6,4 +6,12 @@ info.app.description = ${project.description}
\ No newline at end of file
+info.build.date      = ${timestamp}
+
+## Default values for local testing. Will be ignored in war builds
```
I assume these properties were removed on purpose as they are not required anymore

SrdjanStevanetic replied:
Previously somehow it did not work, but now it works,so I removed them.

gsergiu commented on entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCompleteValidator.java:
```diff
@@ -68,23 +70,27 @@ public boolean isValid(Entity entity, ConstraintValidatorContext context) {
         boolean returnValueLocal = true;
 
-        Class<?> fieldType = field.getType();
+        Class<?> fieldJavaType = field.getType();
```
better use the field object itself as it provides more information about the field, this might be required especially for nested objects

SrdjanStevanetic replied:
Improved according to the new changes.

gsergiu reviewed (commented):
please address the review comments

SrdjanStevanetic committed:
improved the validation of the entity fields and the corresponding tests

SrdjanStevanetic committed:
git-code-format-maven-plugin changed back, it works outside eclipse with
"mvn verify"

SrdjanStevanetic committed:
some additional automated code formatting

SrdjanStevanetic committed:
removed the minContentCount property from the entity fields enumeration
types

ikattey committed:
EA-2851: Improve EuropeanaProxy validation during creation and updates

ikattey committed:
EA-2851: Remove unused code

ikattey committed:
EA-2851: Code cleanup

- removed unused code
- improved HTTP error response message on validation failure
